src/lib/shiorif.ts

```diff
@@ -0,0 +1,226 @@
+import * as ShioriJK from "shiorijk";
+import {ShioriTransaction} from "shiori_transaction";
+import {EventEmitter} from "events";
+import {Shiori} from "shioriloader";
+import {ShioriConverter} from "shiori_converter";
+
+/** The convenient SHIORI Shared Library Interface */
+export class Shiorif extends EventEmitter {
+  /**
+   * convert array values to Reference* hash
+   * @param headersArray header array values
+   * @return headers hash
+   */
+  static referencesFromArray(headersArray: string[]) {
+    const headers: {[name: string]: string} = {};
+    headersArray.forEach((header, index) => {
+      if (header != null) headers[`Reference${index}`] = header;
+    });
+    return headers;
+  }
+
+  private readonly _shiori: Shiori;
+  private readonly _requestParser: ShioriJK.Shiori.Request.Parser;
+  private readonly _responseParser: ShioriJK.Shiori.Response.Parser;
```

Comment:
__[tslint]__
* variable name must be in camelcase or uppercase

<!-- Comment by [SideCI](https://sideci.com) -->